pkg/provisioners/gcppubsub/dispatcher/receiver/receiver.go

```diff
@@ -76,13 +80,21 @@ func (r *Receiver) sendEventToTopic(channel provisioners.ChannelReference, messa
 		return errors.New("status.internal is blank")
 	}
 
-	psr, err := r.createPubSubReceiver(ctx, pcs)
-	if err != nil {
-		logging.FromContext(ctx).Info("Unable to create pubSubReceiver", zap.Error(err))
-		return err
+	cached := r.cache.Get(pcs.Topic)
+	if cached == nil {
```

Comment:
I'm little confused on this logic here. If it's been evicted from the cache (aka, not authoritative source, just the cache), it seems like we'll then try to create the topic again (is it idempotent?). Seems like if we take a cache miss, we'd go to authoritative source and see if the topic exists already <.... time passes... dig through libraries >...

So, I think what's happening is that we just create a reference to the topic, not actually create the topic in createPubSubReceiver, so the fact that it's not in the cache, just means we create a reference to the topic?
Just want to make sure I followed through the calls correctly :)

Reply:
Yes, creating the `pubSubReceiver` essentially creates a reference to the topic (under the hoods it makes a struct, but does not run any methods on it). Then when we call `psr.topic.Publish` it starts all the background work. Which should eventually be stopped by calling `psr.Stop`, which happens on cache eviction. 

Added a comment in `createPubSubReceiver` that tries to explain this, let me know if there should be more comments.